lib/graphql/types/iso_8601_date_time.rb

```diff
@@ -32,7 +32,11 @@ def self.time_precision=(value)
       # @param value [DateTime]
       # @return [String]
       def self.coerce_result(value, _ctx)
-        value.iso8601(time_precision)
+        if value.is_a?(DateTime)
```

Comment:
How will this handle an `ActiveSupport::TimeWithZone`? I don't think it's tested, but I can imagine it would stop working here. Would you mind checking whether it will be affected by this? 

Another approach would be to use `if value.is_a?(Date)` to identify the special case, instead. (🤔 but maybe DateTime _extends_ Date?)

Reply:
That's a valid point! What if we check method _arity_ instead of sticking with exact classes? It should cover all the cases we might face in future (e.g., new Date-ish classes)

Reply:
@rmosolgo Could you please take another look? 🙂